### PR TITLE
Codegen

### PR DESCRIPTION
**Pull request by ian-hoyle**
sbt graphqlCodegen integrated into project 
The https://github.com/Jarlakxen/drunk GraphQLClient and GraphQLCursor has been changed to allow use  of any connection to the GraphQL Server so we can use 
GenericApiGatewayClient https://github.com/rpgreen/apigateway-generic-java-sdk

### CODE REVIEW

File: app/controllers/ViewCollectionsController.scala [hunk 1]
```diff
@@ -1,52 +1,42 @@
...
-                   }
-                }
-           }
-           """
-
-    appSyncClient.query[CollectionsResult](getCollectionsDoc).result.map(result => result match {
-      case Right(r) => Ok(views.html.showCollections(r.data.getCollections.collections))
+    val appSyncClient = client.graphqlClient(List(header))
+
+    //        appSyncClient.query[CreateConsignment.createCon.Data, CreateConsignment.createCon.Variables](CreateConsignment.createCon.document,
```
Comment (TomJKing) [inline app/controllers/ViewCollectionsController.scala:27]: Do we need this commented out code

File: app/graphql/mutations/CreateConsignment.graphql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+mutation createCon ($name: String!, $seriesId: Int!){
```
Comment (TomJKing) [inline app/graphql/mutations/CreateConsignment.graphql]: Maybe change mutation name to createConsignment

File: app/graphql/tdr/TdrGraphQLClient.scala [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+    val result = execute(doc, Some(variables), operationName)
+    result.flatMap { case (status, body) => Future.fromTry(extractErrorOrData(body, status)) }
+  }
+
+  def schema(implicit ec: ExecutionContext): Future[IntrospectionSchema] =
+    execute[Json, Nothing](introspectionQuery, None, None)(null, ec)
+      .flatMap {
+        case (_, json) => Future.fromTry(IntrospectionParser.parse(json))
+      }
+
```
Comment (TomJKing) [inline app/graphql/tdr/TdrGraphQLClient.scala]: Should delete unnecessary lines

File: app/views/showCollections.scala.html [hunk 4]
```diff
@@ -1,16 +1,17 @@
-@(collections:Seq[model.TdrCollection])
```
Comment (TomJKing) [inline app/views/showCollections.scala.html:1]: Should this be renamed as "showConsignments"

File: conf/schema.graphql [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+type File {
+  id: UUID!
+  path: String!
+  consignmentId: Int!
+  fileStatus: FileStatus!
+  pronomId: String
+  fileSize: Int!
+  lastModifiedDate: String!
```
Comment (TomJKing) [inline conf/schema.graphql]: Is this the correct type for lastModifiedDate with the changes made to implement a date types?

File: app/controllers/ViewCollectionsController.scala (hunk 1 above)
Comment (ian-hoyle) [reply app/controllers/ViewCollectionsController.scala:27]: Left for example of mutation and drunk. Can be removed once a real mutation used

File: app/views/showCollections.scala.html (hunk 4 above)
Comment (ian-hoyle) [reply app/views/showCollections.scala.html:1]: done

File: app/graphql/tdr/TdrGraphQLClient.scala (hunk 3 above)
Comment (ian-hoyle) [reply app/graphql/tdr/TdrGraphQLClient.scala]: removed

File: app/graphql/mutations/CreateConsignment.graphql (hunk 2 above)
Comment (ian-hoyle) [reply app/graphql/mutations/CreateConsignment.graphql]: changed

**Pull request merged**